### ISSUE TITLE
parser error on clang 18.1.8

### ISSUE THREAD

**Issue by memory-thrasher** [llvm/llvm-project#97217]
I compiled clang from the latest tag (because I didn't want to do an OS upgrade just to upgrade clang) following this:
https://llvm.org/docs/GettingStarted.html#getting-the-source-code-and-building-llvm
except I used single branch mode to fetch the most recent release commit rather than master:
`git clone https://github.com/llvm/llvm-project.git --single-branch -b 'release/18.x' --depth 1`

`cmake -S llvm -B build -G 'Unix Makefiles' -DLLVM_ENABLE_PROJECTS='clang;clang-tools-extra;lldb;lld' -DCMAKE_BUILD_TYPE=Release`

I used make to install it to the default `/usr/local` prefix, leaving the ubuntu repo clang in `/usr`. The resulting binaries are no longer able to compile my project. My project is accessible here:
https://github.com/memory-thrasher/WITE
but I'm attaching the single header file that is causing the crash with a minimal `main.cpp` that includes it. I'm also attaching the console output as a txt file because I personally don't like scrolling past a wall of logs when looking at someone else's issue. Github wouldn't let me upload without appending .txt to each name.

[stderr.txt](https://github.com/user-attachments/files/16045088/stderr.txt)
[build.sh.txt](https://github.com/user-attachments/files/16045092/build.sh.txt)
[callback.hpp.txt](https://github.com/user-attachments/files/16045093/callback.hpp.txt)
[main.cpp.txt](https://github.com/user-attachments/files/16045094/main.cpp.txt)
[main-6089cf.cpp.txt](https://github.com/user-attachments/files/16045095/main-6089cf.cpp.txt)
[main-6089cf.sh.txt](https://github.com/user-attachments/files/16045096/main-6089cf.sh.txt)

I created /usr/local/bin/clang++-18 as a symlink to ./clang to make clear here the version difference. The bug also occurs with clang invoked as clang++

## Broken clang:
```
clang++-18 --std=c++20 -fPIC -g -Werror -Wall main.cpp -c -o main.o

$ clang++-18 --version
clang version 18.1.8 (https://github.com/llvm/llvm-project.git 3b5b5c1ec4a3095ab096dd780e84d7ab81f3d7ff)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/bin
```

## Working clang:
```
clang++-16 --std=c++20 -fPIC -g -Werror -Wall main.cpp -c -o main.o

$ clang++-16 --version
Ubuntu clang version 16.0.0-++20221021052626+7dd2f4bc009d-1~exp1~20221021172738.418
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

## Question
Is 18.1.8 meant to be stable or have I downloaded a beta by mistake? This seems like a beta kind of problem. If so, what is the latest stable? latest 17.x?

**Comment by EugeneZelenko** [comment]
18.1.x is latest release.

**Comment by DimitryAndric** [comment]
This is a duplicate of #63782, and is fixed with #80594 (in 2c2d291b4568381999442e47fc77f949f19be0bc). It is a regression from 17.x, but the fix did not get merged back into the 18.x branch.
